config/secret_unprotected.go

```diff
@@ -0,0 +1,93 @@
+package config
+
+import (
+	"bytes"
+	"unsafe"
+)
+
+type unprotectedSecretImpl struct{}
+
+func (*unprotectedSecretImpl) Container(secret []byte) secretContainer {
+	return &unprotectedSecretContainer{buf: newUnlockedBuffer(secret)}
+}
+
+func (*unprotectedSecretImpl) EmptyBuffer() SecretBuffer {
+	return &unlockedBuffer{}
+}
+
+func (*unprotectedSecretImpl) Wipe(secret []byte) {
+	for i := range secret {
+		secret[i] = 0
+	}
+}
+
+type unlockedBuffer struct {
+	content []byte
+}
+
+func newUnlockedBuffer(secret []byte) *unlockedBuffer {
+	return &unlockedBuffer{bytes.Clone(secret)}
+}
+
+func (lb *unlockedBuffer) Size() int {
+	return len(lb.content)
+}
+
+func (lb *unlockedBuffer) Grow(_ int) {
+	// The underlying byte-buffer will grow dynamically
+}
+
+func (lb *unlockedBuffer) Bytes() []byte {
+	return lb.content
+}
+
+func (lb *unlockedBuffer) TemporaryString() string {
+	return unsafe.String(&lb.content[0], len(lb.content))
+}
+
+func (lb *unlockedBuffer) String() string {
+	return string(lb.content)
```

Comment:
why one string method to be used over another? Is it possible to leave just one?

Reply:
We need two as in the `TemporaryString` method the string will not be usable after `Destroy` as it is just a "string pointer" to the byte-array-memory region. Therefore we also provide `String` which is a copy to an immutable string. 

The former method is preferred for cases where you authenticate in a function and then can go from there as it will not expose the secret information in the protected case. The unprotected-implementation then also has to serve the interface...